Code.gs

```diff
@@ -99,196 +98,192 @@ function DeleteAllTriggers(){
+    targetCalendar = CalendarApp.createCalendar(targetCalendarName);
+    targetCalendar.setSelected(true); //Sets the calendar as "shown" in the Google Calendar UI
   }
-
+  targetCalendarId = targetCalendar.getId()
+  Logger.log("Working on calendar: " + targetCalendar.getName() + ", ID: " + targetCalendarId)
+  
   if (emailWhenAdded && email == "")
     throw "[ERROR] \"emailWhenAdded\" is set to true, but no email is defined";
   //----------------------------------------------------------------
-
+  
   //------------------------ Parse events --------------------------
-  var feedEventIds=[];
-
+  var icsEventIds=[];
+  
   //Use ICAL.js to parse the data
   var jcalData = ICAL.parse(response);
   var component = new ICAL.Component(jcalData);
   vtimezone = component.getFirstSubcomponent("vtimezone");
-  if (vtimezone != null)
-    ICAL.TimezoneService.register(vtimezone);
+  if (vtimezone == null){
+    tzid = "GMT";
+  }else{
+    tzid = vtimezone.getFirstPropertyValue("tzid") || 'GMT';
+  }
   
   //Map the vevents into custom event objects
-  var events = component.getAllSubcomponents("vevent").map(ConvertToCustomEvent);
-  events.forEach(function(event){ feedEventIds.push(event.id); }); //Populate the list of feedEventIds
+  var icsEvents = component.getAllSubcomponents("vevent").map(ConvertToCustomEvent);
+  icsEvents.forEach(function(event){ icsEventIds.push(event.id); }); //Populate the list of icsEventIds
   //----------------------------------------------------------------
   
-  //------------------------ Check results -------------------------
-  Logger.log("# of events: " + events.length);
-  for each (var event in events){
-    Logger.log("Title: " + event.title);
-    Logger.log("Id: " + event.id);
-    Logger.log("Description: " + event.description);
-    Logger.log("Start: " + event.startTime);
-    Logger.log("End: " + event.endTime);
-
-    for each (var reminder in event.reminderTimes)
-      Logger.log("Reminder: " + reminder + " seconds before");
-
-    Logger.log("");
-  }
-  //----------------------------------------------------------------
-
   if(addEventsToCalendar || removeEventsFromCalendar){
-    var calendarEvents = targetCalendar.getEvents(new Date(2000,01,01), new Date( 2100,01,01 ))
-    var calendarFids = []
+    var calendarEvents = Calendar.Events.list(targetCalendarId).items;
+    var calendarEventsIds = []
+    Logger.log("Calendar currently has " + calendarEvents.length + " existing Events.");
     for (var i = 0; i < calendarEvents.length; i++)
-      calendarFids[i] = calendarEvents[i].getTag("FID");
+      calendarEventsIds[i] = calendarEvents[i].id;
   }
-
+  
   //------------------------ Add events to calendar ----------------
   if (addEventsToCalendar){
-    Logger.log("Checking " + events.length + " Events for creation")
-    for each (var event in events){
-      if (calendarFids.indexOf(event.id) == -1){
-        var resultEvent;
-        if (event.isAllDay){
```

Comment:
Why has the `event.isAllDay` check been removed? Am I missing something? Seems like this is a degradation in functionality

Reply:
The check just moved to line 252. Using the API, all day events are created the same way as 'normal' events so there's no need the check after the new event object has been created.